multiview/cluster/mv_k_means.py

```diff
@@ -0,0 +1,262 @@
+        self._k = k
+        self._random_state = random_state
+
+    def _compute_distance(self, X, centers):
+
+        '''
+        Computes the euclidean distance between each sample point
+        in the given view and each cluster centroid.
+
+        Parameters
+        ----------
+        X : array-like, shape (n_samples, n_features)
+            The data from a single view.
+        centers : array_like, shape (n_clusters, n_features)
+            The cluster centroids for a single view.
+
+        Returns
+        -------
+        distances : array-like (n_clusters, n_samples)
+            An array of euclidean distances between each
+            sample point and each cluster centroid.
+
+        '''
+
+        distances = list()
+        for cl in range(self._k):
+            dist = X - centers[cl]
+            dist = np.linalg.norm(dist, axis=1)
+            distances.append(dist)
+
+        distances = np.vstack(distances)
+        return distances
+
+    def fit(self, Xs, patience=5, max_iter=None):
+
+        '''
+        Fit the cluster centroids to the data.
+
+        Parameters
+        ----------
+        Xs : list of array_likes
+            - Xs shape: (2,)
+            - Xs[0] shape: (n_samples, n_features_i)
+            This list must be of size 2, corresponding to the two views of
+            the data. The two views can each have a different number of
+            features, but they must have the same number of samples.
+
+        patience: int, optional (default=5)
+            The number of EM iterations with no decrease in the objective
+            function after which the algorithm will terminate.
+
+        max_iter: int, optional (default=None)
+            The maximum number of EM iterations to run before
+            termination.
+
+        '''
+
+        Xs = check_Xs(Xs)
+
+        if not (isinstance(patience, int) and (patience > 0)):
+            msg = 'patience must be a nonnegative integer'
+            raise ValueError(msg)
+
+        if max_iter is not None:
+            if not (isinstance(max_iter, int) and (max_iter > 0)):
+                msg = 'max_iter must be a positive integer'
+                raise ValueError(msg)
+        else:
+            max_iter = np.inf
+
+        indices = np.random.choice(Xs[1].shape[0], self._k)
+        centers = Xs[1][indices]
+        self._centroids = [None, centers]
+
+        distances = self._compute_distance(Xs[1], centers)
+        parts = np.argmin(distances, axis=0).flatten()
+        partitions = [None, parts]
+        objective = [np.inf, np.inf]
+        iter_stall = 0
+        iter_num = 0
+        entropy = 0
+
+        while(iter_stall < patience and iter_num < max_iter):
+            iter_num += 1
+            view = (iter_num + 1) % 2
+
+            # Switch partitions, maximization, and expectation
+
+            new_centers = list()
+            for cl in range(self._k):
+                mask = (partitions[(view + 1) % 2] == cl)
```

Comment:
Can you add an inline comment here? Apply to other places where there's a decent amount of steps happening on one line (due to a python/numpy trick).